pubsub/google/cloud/pubsub_v1/subscriber/policy/thread.py

```diff
@@ -273,21 +283,26 @@ def dispatch_callback(self, action, kwargs):
             ValueError: If ``action`` isn't one of the expected actions
                 "ack", "drop", "lease", "modify_ack_deadline" or "nack".
         """
-        if action == 'ack':
-            self.ack(**kwargs)
-        elif action == 'drop':
-            self.drop(**kwargs)
-        elif action == 'lease':
-            self.lease(**kwargs)
-        elif action == 'modify_ack_deadline':
-            self.modify_ack_deadline(**kwargs)
-        elif action == 'nack':
-            self.nack(**kwargs)
-        else:
-            raise ValueError(
-                'Unexpected action', action,
-                'Must be one of "ack", "drop", "lease", '
-                '"modify_ack_deadline" or "nack".')
+        batched_commands = collections.defaultdict(list)
+
+        for item in items:
+            batched_commands[item.__class__].append(item)
+
+        _LOGGER.debug('Handling %d batched requests', len(items))
+
+        if batched_commands[base.LeaseRequest]:
+            self.lease(batched_commands.pop(base.LeaseRequest))
+        if batched_commands[base.ModAckRequest]:
+            self.modify_ack_deadline(
+                batched_commands.pop(base.ModAckRequest))
+        # Note: Drop and ack *must* be after lease. It's possible to get both
```

Comment:
Pub/Sub is smart enough that if you ack and modack, it knows what to do (ignore the modack).
Since nacking is really just a modack of 0, though, order does matter (and may actually not be reliable, but that is not the end of the world).

So, leave as is.